After upgrading to gradle plugin 4.1.0-beta01, local engine builds don't work

```diff
diff a/android/build.gradle b/android/build.gradle
9c9
<         classpath 'com.android.tools.build:gradle:3.5.0'
---
>         classpath 'com.android.tools.build:gradle:4.1.0-beta01'
```

Causes `--local-engine` builds to fail with errors like this:

```
* What went wrong:                                                      
Execution failed for task ':app:checkProfileAarMetadata'.               
> Could not resolve all files for configuration ':app:profileRuntimeClasspath'.
   > Could not find any matches for io.flutter:flutter_embedding_profile:+ as no versions of io.flutter:flutter_embedding_profile are available.
     Searched in the following locations:                               
       - https://dl.google.com/dl/android/maven2/io/flutter/flutter_embedding_profile/maven-metadata.xml
       - https://jcenter.bintray.com/io/flutter/flutter_embedding_profile/maven-metadata.xml
       - file:/var/folders/ly/j71v2zl54n74gp72pw876xv800k69x/T/flutter_tools.vSzwCK/flutter_tool_local_engine_repo.PbTS4I/io/flutter/flutter_embedding_profile/maven-metadata.xml
     Required by:                                                       
         project :app                                                   
   > Could not find any matches for io.flutter:arm64_v8a_profile:+ as no versions of io.flutter:arm64_v8a_profile are available.
     Searched in the following locations:                               
       - https://dl.google.com/dl/android/maven2/io/flutter/arm64_v8a_profile/maven-metadata.xml
       - https://jcenter.bintray.com/io/flutter/arm64_v8a_profile/maven-metadata.xml
       - file:/var/folders/ly/j71v2zl54n74gp72pw876xv800k69x/T/flutter_tools.vSzwCK/flutter_tool_local_engine_repo.PbTS4I/io/flutter/arm64_v8a_profile/maven-metadata.xml
     Required by:                                                       
         project :app      
```

Flutter 1.22.0-10.0.pre.88 • channel unknown • unknown source
Framework • revision e10bdbbd9d (23 hours ago) • 2020-09-07 11:40:03 -0700
Engine • revision ac8b9c4c52
Tools • Dart 2.10.0 (build 2.10.0-86.0.dev)

<!-- Thank you for using Flutter!

     If you are looking for support, please check out our documentation
     or consider asking a question on Stack Overflow:
      * https://flutter.dev/
      * https://api.flutter.dev/
      * https://stackoverflow.com/questions/tagged/flutter?sort=frequent

     If you have found a bug or if our documentation doesn't have an answer
     to what you're looking for, then fill out the template below. Please read
     our guide to filing a bug first: https://flutter.dev/docs/resources/bug-reports
-->

## Steps to Reproduce

<!-- You must include full steps to reproduce so that we can reproduce the problem. -->

1. Run `flutter create bug`.
2. Update the files as follows:
build.gradle: `classpath 'com.android.tools.build:gradle:4.0.0'`
gradle-wrapper.properties: `distributionUrl=https\://services.gradle.org/distributions/gradle-6.5.1-all.zip`
3. `flutter build -v --local-engine-src-path /home/hamdi/engine/src --local-engine=android_debug_unopt apk --debug`

**Actual results:** `Could not find any matches for io.flutter:flutter_embedding_debug:+ `

<details>
  <summary>Logs</summary>

```
[ +148 ms] executing: [/home/hamdi/snap/flutter/common/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +33 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 018467cdb13bb6bce5d2d0d98ca754d874706092
[        ] executing: [/home/hamdi/snap/flutter/common/flutter/] git tag --points-at HEAD
[  +16 ms] Exit code 0 from: git tag --points-at HEAD
[   +1 ms] executing: [/home/hamdi/snap/flutter/common/flutter/] git describe --match *.*.* --long --tags
[  +34 ms] Exit code 0 from: git describe --match *.*.* --long --tags
[        ] 1.24.0-7.0.pre-194-g018467cdb1
[  +12 ms] executing: [/home/hamdi/snap/flutter/common/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +15 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/master
[        ] executing: [/home/hamdi/snap/flutter/common/flutter/] git ls-remote --get-url origin
[   +6 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +77 ms] executing: [/home/hamdi/snap/flutter/common/flutter/] git rev-parse --abbrev-ref HEAD
[  +10 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] master
[ +106 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[  +14 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +102 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[   +8 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +76 ms] Skipping pub get: version match.
[ +197 ms] Generating /home/hamdi/bug/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +90 ms] Running Gradle task 'assembleDebug'...
[   +3 ms] gradle.properties already sets `android.enableR8`
[   +5 ms] Using gradle from /home/hamdi/bug/android/gradlew.
[  +12 ms] /home/hamdi/bug/android/gradlew mode: 33261 rwxr-xr-x.
[  +41 ms] Using local engine: /home/hamdi/engine/src/out/android_debug_unopt
           Local Maven repo: /tmp/flutter_tools.DEMHHM/flutter_tool_local_engine_repo.FQHOFM
[   +9 ms] executing: [/home/hamdi/bug/android/] /home/hamdi/bug/android/gradlew -Pverbose=true
-Plocal-engine-repo=/tmp/flutter_tools.DEMHHM/flutter_tool_local_engine_repo.FQHOFM -Plocal-engine-build-mode=debug
-Plocal-engine-out=/home/hamdi/engine/src/out/android_debug_unopt -Ptarget-platform=android-arm -Ptarget=lib/main.dart
-Ptrack-widget-creation=true -Pshrink=true assembleDebug
[ +876 ms] Welcome to Gradle 6.5.1!
[        ] Here are the highlights of this release:
[        ]  - Experimental file-system watching
[        ]  - Improved version ordering
[        ]  - New samples
[        ] For more details see https://docs.gradle.org/6.5.1/release-notes.html
[ +179 ms] Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details
[+8211 ms] > Configure project :app
[        ] WARNING: The option setting 'android.enableR8=true' is deprecated.
[        ] It will be removed in version 5.0 of the Android Gradle plugin.
[        ] You will no longer be able to disable R8
[+6387 ms] FAILURE: Build failed with an exception.
[   +2 ms] * What went wrong:
[        ] Could not determine the dependencies of task ':app:compileDebugJavaWithJavac'.
[        ] > Could not resolve all task dependencies for configuration ':app:debugCompileClasspath'.
[        ]    > Could not find any matches for io.flutter:flutter_embedding_debug:+ as no versions of io.flutter:flutter_embedding_debug are
available.
[        ]      Required by:
[        ]          project :app
[        ]    > Could not find any matches for io.flutter:armeabi_v7a_debug:+ as no versions of io.flutter:armeabi_v7a_debug are available.
[        ]      Required by:
[        ]          project :app
[        ] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to
get full insights.
[   +1 ms] * Get more help at https://help.gradle.org
[        ] BUILD FAILED in 15s
[        ] Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
[        ] Use '--warning-mode all' to show the individual deprecation warnings.
[        ] See https://docs.gradle.org/6.5.1/userguide/command_line_interface.html#sec:command_line_warnings
[ +611 ms] Running Gradle task 'assembleDebug'... (completed in 16.3s)
[  +25 ms] "flutter apk" took 16,899ms.
[  +18 ms] Gradle task assembleDebug failed with exit code 1
[        ] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      buildGradleApp (package:flutter_tools/src/android/gradle.dart:414:7)
           <asynchronous suspension>
           #2      _AndroidBuilderImpl.buildApk (package:flutter_tools/src/android/android_builder.dart:101:7)
           <asynchronous suspension>
           #3      BuildApkCommand.runCommand (package:flutter_tools/src/commands/build_apk.dart:119:5)
           <asynchronous suspension>
           #4      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1079:12)
           <asynchronous suspension>
           #5      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:934:27)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #8      CommandRunner.runCommand (package:args/command_runner.dart:197:13)
           <asynchronous suspension>
           #9      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:284:9)
           <asynchronous suspension>
           #10     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #11     AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #12     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:230:5)
           <asynchronous suspension>
           #13     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:63:9)
           <asynchronous suspension>
           #14     run.<anonymous closure> (package:flutter_tools/runner.dart:61:12)
           <asynchronous suspension>
           #15     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #16     AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #17     runInContext (package:flutter_tools/src/context_runner.dart:70:10)
           <asynchronous suspension>
           #18     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>
           
           
[ +256 ms] ensureAnalyticsSent: 252ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[   +1 ms] Shutdown hooks complete
[   +1 ms] exiting with code 1

```
</details>

<details>
  <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.24.0-8.0.pre.194, on Linux, locale en_US.UTF-8)
    • Flutter version 1.24.0-8.0.pre.194 at /home/hamdi/snap/flutter/common/flutter
    • Framework revision 018467cdb1 (7 days ago), 2020-11-11 02:04:03 -0500
    • Engine revision 81f219c59c
    • Dart version 2.12.0 (build 2.12.0-31.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/hamdi/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_272-8u272-b10-0ubuntu1~18.04-b10)
    • All Android licenses accepted.

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[!] Connected device
    ! No devices available

! Doctor found issues in 2 categories.

```

</details>

--- a/packages/flutter_tools/lib/src/android/gradle.dart
+++ b/packages/flutter_tools/lib/src/android/gradle.dart
@@ -1090,6 +1090,21 @@ Directory _getLocalEngineRepo({
       ),
     );
   }
+  for (final String artifact in <String>['flutter_embedding_$buildMode', '${abi}_$buildMode']) {
+    _createSymlink(
+      globals.fs.path.join(
+        engineOutPath,
+        '$artifact.maven-metadata.xml',
+      ),
+      globals.fs.path.join(
+        localEngineRepo.path,
+        'io',
+        'flutter',
+        artifact,
+        'maven-metadata.xml',
+      ),
+    );
+  }
   return localEngineRepo;
 }
 

--- a/packages/flutter_tools/test/general.shard/android/gradle_test.dart
+++ b/packages/flutter_tools/test/general.shard/android/gradle_test.dart
@@ -1713,8 +1713,10 @@ plugin1=${plugin1.path}
 ''');
       fileSystem.file('out/android_arm/armeabi_v7a_release.pom').createSync(recursive: true);
       fileSystem.file('out/android_arm/armeabi_v7a_release.jar').createSync(recursive: true);
+      fileSystem.file('out/android_arm/armeabi_v7a_release.maven-metadata.xml').createSync(recursive: true);
       fileSystem.file('out/android_arm/flutter_embedding_release.jar').createSync(recursive: true);
       fileSystem.file('out/android_arm/flutter_embedding_release.pom').createSync(recursive: true);
+      fileSystem.file('out/android_arm/flutter_embedding_release.maven-metadata.xml').createSync(recursive: true);
 
       fileSystem.file('android/gradlew').createSync(recursive: true);
 
@@ -1810,8 +1812,10 @@ plugin1=${plugin1.path}
 ''');
       fileSystem.file('out/android_arm64/arm64_v8a_release.pom').createSync(recursive: true);
       fileSystem.file('out/android_arm64/arm64_v8a_release.jar').createSync(recursive: true);
+      fileSystem.file('out/android_arm64/arm64_v8a_release.maven-metadata.xml').createSync(recursive: true);
       fileSystem.file('out/android_arm64/flutter_embedding_release.jar').createSync(recursive: true);
       fileSystem.file('out/android_arm64/flutter_embedding_release.pom').createSync(recursive: true);
+      fileSystem.file('out/android_arm64/flutter_embedding_release.maven-metadata.xml').createSync(recursive: true);
 
       fileSystem.file('android/gradlew').createSync(recursive: true);
 
@@ -1907,8 +1911,10 @@ plugin1=${plugin1.path}
 ''');
       fileSystem.file('out/android_x86/x86_release.pom').createSync(recursive: true);
       fileSystem.file('out/android_x86/x86_release.jar').createSync(recursive: true);
+      fileSystem.file('out/android_x86/x86_release.maven-metadata.xml').createSync(recursive: true);
       fileSystem.file('out/android_x86/flutter_embedding_release.jar').createSync(recursive: true);
       fileSystem.file('out/android_x86/flutter_embedding_release.pom').createSync(recursive: true);
+      fileSystem.file('out/android_x86/flutter_embedding_release.maven-metadata.xml').createSync(recursive: true);
 
       fileSystem.file('android/gradlew').createSync(recursive: true);
 
@@ -2004,8 +2010,10 @@ plugin1=${plugin1.path}
 ''');
       fileSystem.file('out/android_x64/x86_64_release.pom').createSync(recursive: true);
       fileSystem.file('out/android_x64/x86_64_release.jar').createSync(recursive: true);
+      fileSystem.file('out/android_x64/x86_64_release.maven-metadata.xml').createSync(recursive: true);
       fileSystem.file('out/android_x64/flutter_embedding_release.jar').createSync(recursive: true);
       fileSystem.file('out/android_x64/flutter_embedding_release.pom').createSync(recursive: true);
+      fileSystem.file('out/android_x64/flutter_embedding_release.maven-metadata.xml').createSync(recursive: true);
 
       fileSystem.file('android/gradlew').createSync(recursive: true);
 
@@ -2153,8 +2161,10 @@ plugin1=${plugin1.path}
 ''');
       fileSystem.file('out/android_arm/armeabi_v7a_release.pom').createSync(recursive: true);
       fileSystem.file('out/android_arm/armeabi_v7a_release.jar').createSync(recursive: true);
+      fileSystem.file('out/android_arm/armeabi_v7a_release.maven-metadata.xml').createSync(recursive: true);
       fileSystem.file('out/android_arm/flutter_embedding_release.jar').createSync(recursive: true);
       fileSystem.file('out/android_arm/flutter_embedding_release.pom').createSync(recursive: true);
+      fileSystem.file('out/android_arm/flutter_embedding_release.maven-metadata.xml').createSync(recursive: true);
 
       final File manifestFile = fileSystem.file('pubspec.yaml');
       manifestFile.createSync(recursive: true);
@@ -2254,8 +2264,10 @@ plugin1=${plugin1.path}
 ''');
       fileSystem.file('out/android_arm64/arm64_v8a_release.pom').createSync(recursive: true);
       fileSystem.file('out/android_arm64/arm64_v8a_release.jar').createSync(recursive: true);
+      fileSystem.file('out/android_arm64/arm64_v8a_release.maven-metadata.xml').createSync(recursive: true);
       fileSystem.file('out/android_arm64/flutter_embedding_release.jar').createSync(recursive: true);
       fileSystem.file('out/android_arm64/flutter_embedding_release.pom').createSync(recursive: true);
+      fileSystem.file('out/android_arm64/flutter_embedding_release.maven-metadata.xml').createSync(recursive: true);
 
       final File manifestFile = fileSystem.file('pubspec.yaml');
       manifestFile.createSync(recursive: true);
@@ -2356,8 +2368,10 @@ plugin1=${plugin1.path}
 ''');
       fileSystem.file('out/android_x86/x86_release.pom').createSync(recursive: true);
       fileSystem.file('out/android_x86/x86_release.jar').createSync(recursive: true);
+      fileSystem.file('out/android_x86/x86_release.maven-metadata.xml').createSync(recursive: true);
       fileSystem.file('out/android_x86/flutter_embedding_release.jar').createSync(recursive: true);
       fileSystem.file('out/android_x86/flutter_embedding_release.pom').createSync(recursive: true);
+      fileSystem.file('out/android_x86/flutter_embedding_release.maven-metadata.xml').createSync(recursive: true);
 
       final File manifestFile = fileSystem.file('pubspec.yaml');
       manifestFile.createSync(recursive: true);
@@ -2458,8 +2472,10 @@ plugin1=${plugin1.path}
 ''');
       fileSystem.file('out/android_x64/x86_64_release.pom').createSync(recursive: true);
       fileSystem.file('out/android_x64/x86_64_release.jar').createSync(recursive: true);
+      fileSystem.file('out/android_x64/x86_64_release.maven-metadata.xml').createSync(recursive: true);
       fileSystem.file('out/android_x64/flutter_embedding_release.jar').createSync(recursive: true);
       fileSystem.file('out/android_x64/flutter_embedding_release.pom').createSync(recursive: true);
+      fileSystem.file('out/android_x64/flutter_embedding_release.maven-metadata.xml').createSync(recursive: true);
 
       final File manifestFile = fileSystem.file('pubspec.yaml');
       manifestFile.createSync(recursive: true);

